### PR TITLE
Fusion inventory compatibility

### PR DESCRIPTION
**Pull request by stonebuzz**
ADd capabitlity to fields to manage custom fields from fusionInventory

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: inc/inventory.class.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+         //The file doesn't exists, check without the extension (old format)
+         $file           = PLUGIN_FUSIONINVENTORY_XML_DIR;
+         $filename       = $items_id;
+         $file_shortname = strtolower($itemtype)."/".$folder."/".$filename;
+         $file          .= $file_shortname;
+         if (file_exists($file)) {
+            $pxml = @simplexml_load_file($file, 'SimpleXMLElement', LIBXML_NOCDATA);
+         }
+      } else {
+         $pxml = @simplexml_load_file($file, 'SimpleXMLElement', LIBXML_NOCDATA);
```
Comment (trasher) [inline inc/inventory.class.php:105]: If you use `@`, then you must handle xml parsing errors to log if something goes wrong

File: inc/inventory.class.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      }
```
Comment (cedric-anne) [inline inc/inventory.class.php:106]:
```suggestion
      $file = PLUGIN_FUSIONINVENTORY_XML_DIR . strtolower($itemtype) . "/" . $folder . "/" . $items_id;
      if (file_exists($file . '.xml')) {
         $file .= '.xml';
      } else if (!file_exists($file)) {
         return false;
      }

      $pxml = @simplexml_load_file($file, 'SimpleXMLElement', LIBXML_NOCDATA);
```

File: inc/inventory.class.php (hunk 1 above)
Comment (stonebuzz) [reply inc/inventory.class.php]: done

File: inc/inventory.class.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                  if ($db_info->getFromDBByCrit(['computers_id' => $items_id])) {
+
+                     $arrayinventory = unserialize(gzuncompress($db_info->fields['serialized_inventory']));
+                     if (isset($arrayinventory['custom'])) {
+                        if (isset($arrayinventory['custom']['container']['ID'])) {
+                           $customData['container'][0] = $arrayinventory['custom']['container'];
+                        } else {
+                           $customData = $arrayinventory['custom'];
+                        }
+                        self::updateFields($customData, $itemtype, $items_id);
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply inc/inventory.class.php]: done

File: inc/inventory.class.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                  $file = self::loadXMLFile($itemtype, $items_id);
+                  if ($file !== false) {
+                     $arrayinventory = PluginFusioninventoryFormatconvert::XMLtoArray($file);
+                     if (isset($arrayinventory['CUSTOM'])) {
+                        if (isset($arrayinventory['CUSTOM']['CONTAINER']['ID'])) {
+                           $customData['container'][0] = $arrayinventory['CUSTOM']['CONTAINER'];
+                        } else {
+                           $customData['container'] = $arrayinventory['CUSTOM']['CONTAINER'];
+                        }
+                        self::updateFields($customData, $itemtype, $items_id);
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply inc/inventory.class.php]: done

File: inc/inventory.class.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+         $data = [];
+         $data["items_id"] = $items_id;
+         $data["itemtype"] = $itemtype;
+         $data["plugin_fields_containers_id"] = $dataContainers['ID'];
+         foreach ($dataContainers['FIELDS'] as $key => $value) {
+            $data[strtolower($key)] = $value;
+         }
+         $container->updateFieldsValues($data, $itemtype, false);
+      }
+   }
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply inc/inventory.class.php]: done

File: inc/inventory.class.php (hunk 2 above)
Comment (stonebuzz) [reply inc/inventory.class.php]: done

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**